.gitlab-ci.yml

```diff
@@ -21,7 +21,7 @@ variables:
   RUSTY_CACHIER_SINGLE_BRANCH:     master
   RUSTY_CACHIER_DONT_OPERATE_ON_MAIN_BRANCH: "true"
   # paritytech/contracts-ci-linux:production defaults to nightly toolchain, default rusty-cachier to it too
-  RUSTY_CACHIER_TOOLCHAIN:         nightly
+  RUSTY_CACHIER_TOOLCHAIN:         stable
```

Comment:
I have  no idea what this env variable actually does. It didn't change the toolchain being used to compile because that needed to be changed at other places.

Reply:
> This variable is used by CI tool [rusty-cachier](https://gitlab.parity.io/parity/infrastructure/ci_cd/rusty-cachier/client) which provides reusable Rust cache for Cargo.

I figured that much but what does this actually mean?

Reply:
> We'll need to update the Docker image in `paritytech/scripts` after this, eh

Yes, the default toolchain for the CI image is [nightly](https://github.com/paritytech/scripts/blob/master/dockerfiles/contracts-ci-linux/Dockerfile#L51). But it could be overridden in the `before_script:` section as it was done below.

> I figured that much but what does this actually mean?

`rusty-cachier` related environment variables are described [here](https://gitlab.parity.io/parity/infrastructure/ci_cd/rusty-cachier/client/-/blob/master/README.md#environment-variables). `RUSTY_CACHIER_TOOLCHAIN` [is used to select the toolchain](https://gitlab.parity.io/parity/infrastructure/ci_cd/rusty-cachier/client/-/blob/5fb5a4d86005b825e200528b960ce7e14b09c921/lib/cacheresolver.rb#L31) to construct a cache key and defaults to stable. Toolchain version is used as a part of the cache key because different Rust versions have incompatible cargo_target_dir caches.

Reply:
If all pipeline jobs are going to use the stable toolchain, the variable could be removed.

Reply:
I was more referring to the line with `paritytech/contracts-ci-linux:production` (which wasn't touched in this PR).

@rcny not **all** jobs are going to use the same toolchain, some are still gonna use `nightly`. What should we do in this case?

Reply:
Add `RUSTY_CACHIER_TOOLCHAIN: nightly` under the `variables:` keyword at the specific jobs' level.